fixbackend/domain_events/consumers.py

```diff
@@ -0,0 +1,92 @@
+#  Copyright (c) 2023. Some Engineering
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Affero General Public License for more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import logging
+from datetime import timedelta
+from typing import Any
+
+from fixcloudutils.redis.event_stream import Json, MessageContext, RedisStreamListener
+from fixcloudutils.service import Service
+from httpx import AsyncClient, BasicAuth, Request
+from redis.asyncio import Redis
+
+from fixbackend.config import Config
+from fixbackend.domain_events.converter import converter
+from fixbackend.domain_events.events import UserRegistered
+
+log = logging.getLogger(__name__)
+
+
+class CustomerIoEventConsumer(Service):
+    def __init__(
+        self,
+        http_client: AsyncClient,
+        config: Config,
+        readwrite_redis: Redis,
+        stream_name: str,
+    ) -> None:
+        self.http_client = http_client
+        self.site_id = config.customerio_site_id
+        self.api_key = config.customerio_api_key
+        self.customerio_baseurl = config.customerio_baseurl
+        self.listener = RedisStreamListener(
+            readwrite_redis,
+            stream_name,
+            group="domainevent-customerio",
+            listener=config.instance_id,
```

Comment:
FYI: added an option to define the instance id via env var and provide it in k8s.